### PR TITLE
frontend-utils: Move a bunch of backends from Desktop

### PR DESCRIPTION
**Pull request by Dinnerbone**
Most of this has been duplicated over on Android, but then diverged as each codebase went its own way. This lets us bring it all together again, and makes maintaining things so much easier.

### CODE REVIEW

**Review by sleepycatcoding — APPROVED** [review]
Looks good to me.

File: desktop/src/executor.rs [hunk 1]
```diff
@@ -129,7 +132,7 @@ impl TaskHandle {
     );
 }
 
-pub struct WinitAsyncExecutor {
+pub struct WinitAsyncExecutor<R: PollRequester> {
```
Comment (Lord-McSweeney) [inline desktop/src/executor.rs]: Should this still be called `WinitAsyncExecutor`?

Comment (sleepycatcoding) [reply desktop/src/executor.rs]: It is renamed in a later commit: https://github.com/ruffle-rs/ruffle/pull/15949/commits/3ace44006b1fba5114bba0971848a7db98fd1c08

File: frontend-utils/src/backends/navigator.rs [hunk 2]
```diff
@@ -0,0 +1,890 @@
+//! Navigator backend for web
```
[the remark this answers is not in the document]
Comment (Dinnerbone) [reply frontend-utils/src/backends/navigator.rs:1]: Yes :D I'll remove it though

**Pull request merged**